onadata/libs/pagination.py

```diff
@@ -57,3 +62,73 @@ def paginate_queryset(self, queryset, request, view, count=None):
 
         self.request = request
         return list(self.page)
+
+
+def generate_pagination_headers(
+    request: Request, no_of_records: int, current_page_size: Union[int, str],
+    current_page: Union[int, str]
+) -> dict:
+    url = urlparse(request.build_absolute_uri())
+    base_url = f"{url.scheme}://{url.netloc}{url.path}?"
+    if url.query:
+        query_params = parse_qs(url.query)
+        query_string = None
+        for param_key, param_values in query_params.items():
+            if param_key not in ['page_size', 'page']:
+                param_value = ','.join(param_values)
+                query = f'{param_key}={param_value}&'
+                if not query_string:
+                    query_string = query
+                else:
+                    query_string += query
+
+        base_url += f'{query_string}'
+
+    next_page_url = None
+    prev_page_url = None
+    first_page_url = None
+    last_page_url = None
+    links = []
+
+    if isinstance(current_page, str):
+        try:
+            current_page = int(current_page)
+        except ValueError:
+            return
+
+    if isinstance(current_page_size, str):
+        try:
+            current_page_size = int(current_page_size)
+        except ValueError:
+            return
+
+    if (current_page * current_page_size) < no_of_records:
+        next_page_url = (
+            f"{base_url}page={current_page + 1}&"
```

Comment:
I see the parent class has `get_next_link` Could we re-use that functionality here and the get previous link
Ref https://github.com/encode/django-rest-framework/blob/master/rest_framework/pagination.py#L271

Reply:
This function isn't really in the Pagination class at the moment... I can add it to the Pagination class though

Reply:
i meant in the `PageNumberPagination` class